src/vm/assemblynative.cpp

```diff
@@ -1331,7 +1331,10 @@ INT_PTR QCALLTYPE AssemblyNative::GetLoadContextForAssembly(QCall::AssemblyHandl
     // We should have a load context binder at this point.
     _ASSERTE(pOpaqueBinder != nullptr);
 
-    if (!AreSameBinderInstance(pTPABinder, pOpaqueBinder))
+    // the TPA binder uses the default ALC
+    // WinRT assemblies (bound using the WinRT binder) don't actually have an ALC,
+    // so treat them the same as if they were loaded into the TPA ALC in this case.
+    if (!AreSameBinderInstance(pTPABinder, pOpaqueBinder) && !AreSameBinderInstance(pCurDomain->GetWinRtBinder(), pOpaqueBinder))
```

Comment:
Is this the correct behavior? I wonder if `WinRtBinder` shouldn't just return `null` for `GetLoadContextForAssembly`

@jkotas What do you think?

Reply:
I think that this fix is fine.

Reply:
This would imply this assertion would pass
```C#
Assert.AreEqual(AssemblyLoadContext.Default.LoadFromAssembly(WinRtAssembly.GetName()), WinRtAssembly);
```

Reply:
The call `AssemblyLoadContext.Default.LoadFromAssembly(WinRtAssembly.GetName())` itself will fail to load a WinRT assembly.

Reply:
WinMDs have to be loaded very particularly to make sure they're loaded via the WinRT binder.

Reply:
So in my humble opinion, the function should return null.

Reply:
Maybe it doesn't matter, because normal assemblies will not depend on them.

Reply:
> the function should return null

Which function?

Reply:
> Which function?

`AssemblyNative::GetLoadContextForAssembly(Assembly)`

Reply:
`AssemblyNative::GetLoadContextForAssembly` will return null for WinRT assemblies with this change.

Reply:
Hmm! I should read code more thoroughly. `AssemblyLoadContext.GetLoadContext(Assembly assembly)` converts `null` as `Default`.  I assumed that conversion was inside the C++ function.

```C#
                IntPtr ptrAssemblyLoadContext = GetLoadContextForAssembly(rtAsm);
                if (ptrAssemblyLoadContext == IntPtr.Zero)
                {
                    // If the load context is returned null, then the assembly was bound using the TPA binder
                    // and we shall return reference to the active "Default" binder - which could be the TPA binder
                    // or an overridden CLRPrivBinderAssemblyLoadContext instance.
                    loadContextForAssembly = AssemblyLoadContext.Default;
                }
```

But I guess the comment notes that precedence has already been established...

I really wanted `AssemblyLoadContext.GetLoadContext(Assembly winRtAssembly) == null`